firmware/thruna/PDM/src/app/app_powerManager.c

```diff
@@ -1,49 +1,131 @@
-            [EFUSE_CHANNEL_EMETER] = false,
-            [EFUSE_CHANNEL_AUX] = false,
-            [EFUSE_CHANNEL_DRS] = false,
-            [EFUSE_CHANNEL_FAN] = true,
-            [EFUSE_CHANNEL_DI_LHS] = true,
-            [EFUSE_CHANNEL_DI_RHS] = true,
+        .retry_configs = {
+            [EFUSE_CHANNEL_AIR] = {true, RETRY_PROTOCOL_AIR, 1},
+            [EFUSE_CHANNEL_LVPWR] = {true, RETRY_PROTOCOL_LV, 1},
+            [EFUSE_CHANNEL_EMETER] = {false, RETRY_PROTOCOL_EMETER, 3},
+            [EFUSE_CHANNEL_AUX] = {false, RETRY_PROTOCOL_AUX, 3},
+            [EFUSE_CHANNEL_DRS] = {false, RETRY_PROTOCOL_DRS, 3},
+            [EFUSE_CHANNEL_FAN] = {true, RETRY_PROTOCOL_FANS, 3},
+            [EFUSE_CHANNEL_DI_LHS] = {true, RETRY_PROTOCOL_DI_LHS, 1},
+            [EFUSE_CHANNEL_DI_RHS] = {true, RETRY_PROTOCOL_DI_RHS, 1},
         }
     }
 };
 
+void app_powerManager_init()
+{
+    for (int efuse = 0; efuse < NUM_EFUSE_CHANNELS; efuse++)
+    {
+        retry_data[efuse].retry_state          = RETRY_STATE_OFF;
+        retry_data[efuse].retry_attempts       = 0;
+        retry_data[efuse].timer_attempts       = 0;
+        retry_data[efuse].timer_attempts_limit = 3;
+        retry_data[efuse].current_sum          = 0.0;
+    }
+}
+
 void app_powerManager_setState(PowerManagerState state)
 {
     current_power_state = state;
 
     for (int efuse = 0; efuse < NUM_EFUSE_CHANNELS; efuse++)
     {
-        io_efuse_setChannel(efuse, power_states_config[state].efuses[efuse]);
+        io_efuse_setChannel(efuse, power_states_config[state].retry_configs[efuse].efuse_state);
+    }
+}
+
+void app_powerManager_check_efuses(PowerManagerState state)
+{
+    int efuse = 0;
+
+    while (efuse < NUM_EFUSE_CHANNELS)
+    {
+        RetryData         *efuse_retry_data   = &retry_data[efuse];
+        const RetryConfig *efuse_retry_config = &power_states_config[state].retry_configs[efuse];
+
+        // Check if timer attempts reached the limit
+        if (efuse_retry_data->timer_attempts == efuse_retry_data->timer_attempts_limit)
+        {
+            efuse_retry_data->retry_state = RETRY_STATE_EXPIRED;
+        }
+
+        switch (efuse_retry_data->retry_state)
+        {
+            case RETRY_STATE_WAITING:
+                break;
+            case RETRY_STATE_RUNNING:
+            {
+                efuse_retry_data->current_sum += io_efuse_getChannelCurrent(efuse);
+                efuse_retry_data->timer_attempts++;
+                break;
+            }
+            case RETRY_STATE_EXPIRED:
+            {
+                float avg = efuse_retry_data->current_sum / (float)efuse_retry_data->timer_attempts_limit;
+                if (avg <= FAULT_CURRENT_THRESHOLD)
+                {
+                    if (efuse_retry_data->retry_attempts == efuse_retry_config->retry_attempts_limit)
+                    {
+                        app_canTx_PDM_EfuseFault_set(true);
+                        return;
+                    }
```

Comment:
You can probably have this check at the top of the for loop, since there's really no need to stay do the rest of the checks if its already expired.